libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -479,76 +497,37 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync recordIds:(NSArray*)recordIds in
         [self.store upsertEntries:@[record] toSoup:soupName];
         
         // Next
-        [self syncUpOneEntry:sync recordIds:recordIds index:i+1 updateSync:updateSync failRest:failRest];
+        [self syncUpOneEntry:sync recordIds:recordIds index:i+1 updateSync:updateSync failBlock:failBlock];
     };
-
+    
     // Create handler
-    SFRestDictionaryResponseBlock completeBlockCreate = ^(NSDictionary *d) {
+    void (^completeBlockCreate)(NSDictionary *) = ^(NSDictionary *d) {
         // Replace id with server id during create
         record[kId] = d[kSyncManagerLObjectId];
         completeBlockUpdate(d);
     };
-
-    SFRestRequest* request;
-    id completeBlock;
+    
+    void (^completeBlock)(NSDictionary *);
     switch(action) {
-        case kSyncManagerActionCreate:
-            request = [[SFRestAPI sharedInstance] requestForCreateWithObjectType:objectType fields:fields];
+        case SFSyncServerTargetActionCreate:
             completeBlock = completeBlockCreate;
             break;
-        case kSyncManagerActionUpdate:
-            request = [[SFRestAPI sharedInstance] requestForUpdateWithObjectType:objectType objectId:objectId fields:fields];
+        case SFSyncServerTargetActionUpdate:
             completeBlock = completeBlockUpdate;
             break;
-        case kSyncManagerActionDelete:
-            request = [[SFRestAPI sharedInstance] requestForDeleteWithObjectType:objectType objectId:objectId];
+        case SFSyncServerTargetActionDelete:
             completeBlock = completeBlockDelete;
             break;
-        case kSyncManagerActionNone: /* caught by if (action == kSyncManagerActionNone) above */ break;
+        default:
+            // Action is unsupported here.  Move on.
+            [self log:SFLogLevelInfo format:@"%@ unsupported action with value %d.  Moving to the next record.", NSStringFromSelector(_cmd), action];
+            [self syncUpOneEntry:sync recordIds:recordIds index:i+1 updateSync:updateSync failBlock:failBlock];
+            return;
     }
-
-    // Send request
-    [self sendRequestWithSmartSyncUserAgent:request failBlock:failRest completeBlock:completeBlock];
-}
-
-- (void)sendRequestWithSmartSyncUserAgent:(SFRestRequest *)request failBlock:(SFRestFailBlock)failBlock completeBlock:(id)completeBlock {
-    //[request setHeaderValue:[SFRestAPI userAgentString:kSmartSync] forHeaderName:kUserAgent];
-    [self.restClient sendRESTRequest:request failBlock:failBlock completeBlock:completeBlock];
-}
-
-- (void)isNewerThanServer:(SFSyncState*)sync recordIds:(NSArray*)recordIds index:(NSUInteger)i record:(NSMutableDictionary*)record action:(SFSyncManagerAction)action updateSync:(SyncUpdateBlock)updateSync failRest:(SFRestFailBlock)failRest {
```

Comment:
Yes.  The `fetch` method decouples the comparison process, simply returning the relevant dates to the sync manager, which then implements the check.  That way we can support other kinds of checks in the future, without modifying the targets.